frontend/javascripts/oxalis/view/left-border-tabs/modals/add_volume_layer_modal.tsx

```diff
@@ -17,11 +17,67 @@ import {
 import {
   getAllReadableLayerNames,
   getVolumeTracingLayers,
-  validateReadableLayerName,
 } from "oxalis/model/accessors/volumetracing_accessor";
+import messages from "messages";
 import InputComponent from "oxalis/view/components/input_component";
 import api from "oxalis/api/internal_api";
 import Toast from "libs/toast";
+
+export type ValidationResult = { isValid: boolean; message: string };
+export function checkForLayerNameDuplication(
+  readableLayerName: string,
+  allReadableLayerNames: string[],
+): ValidationResult {
+  const layerNameDoesNotExist = _.countBy(allReadableLayerNames)[readableLayerName] <= 0;
+  return {
+    isValid: layerNameDoesNotExist,
+    message: layerNameDoesNotExist ? "" : messages["tracing.volume_layer_name_duplication"],
+  };
+}
+
+export function checkLayerNameForInvalidCharacters(readableLayerName: string): ValidationResult {
+  const uriSaveCharactersRegex = /[0-9a-zA-Z-._~]+/g;
+  // Removing all URISaveCharacters from readableLayerName. The left over chars are all invalid.
+  const allInvalidChars = readableLayerName.replace(uriSaveCharactersRegex, "");
+  const allUniqueInvalidCharsAsSet = new Set(...allInvalidChars);
+  const allUniqueInvalidCharsAsString = "".concat(...allUniqueInvalidCharsAsSet.values());
+  const isValid = allUniqueInvalidCharsAsString.length === 0;
+  return {
+    isValid,
+    message: isValid
+      ? ""
+      : messages["tracing.volume_layer_name_includes_invalid_characters"](
+          allUniqueInvalidCharsAsString,
+        ),
+  };
+}
+
+export function validateReadableLayerName(
+  readableLayerName: string,
+  allReadableLayerNames: string[],
+  nameNotToCount?: string,
+): ValidationResult {
+  if (nameNotToCount) {
+    // If the given nameNotToCount is already once within the allReadableLayerNames. But as this name is now being renamed and having
+    // the same name as the previous name given by nameNotToCount should be still allowed, this name is removed from the array once.
+    // Nevertheless, additional duplicated of nameNotToCount are counted as a duplication.
```

Comment:
There seem to be a couple of missing words. Could you re-check this paragraph? :)